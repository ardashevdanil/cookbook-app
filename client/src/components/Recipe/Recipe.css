.Recipe {
  height: 440px;
  margin: 81px 0;
  display: flex;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
  background-color: #f8f8f8;
}

.Recipe__img {
  height: 100%;
  width: 520px;
}

.Recipe__img>img {
  height: 100%;
}

.Recipe__description {
  padding: 32px 40px;
  background-color: #f8f8f8;
}

.Recipe__heading {
  font-family: 'Playfair Display', serif;
  font-size: 33px;
  font-weight: 900;
  color: #055555;
}

.Recipe__stats {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  margin: 18px 0 30px;
  display: flex;
  align-items: center;
}

.Recipe__stats>svg {
  height: 14px;
}

.Recipe__text {
  height: calc(100% - 150px);
  font-size: 15px;
  line-height: 19px;
  overflow: auto;
}

.Recipe__heading_small {
  font-size: 16px;
  margin: 18px 0 20px;
}

@media only screen and (max-width: 1120px) {
  .Recipe {
    display: block;
    height: auto;
    box-shadow: none;
    margin: 0;
  }

  .Recipe__img {
    height: 64.7vw;
    width: 100%;
    overflow: hidden;
  }

  .Recipe__img>img {
    width: 100%;
    height: auto;
  }

  .Recipe__description {
    padding: 4vw 6.9vw;
  }

  .Recipe__text {
    font-size: 4.4vw;
    line-height: 5.9vw;
  }

  .Recipe__stats {
    font-size: 5vw;
    margin: 5.6vw 0 9.4vw;
  }

  .Recipe__stats>svg {
    height: 4.4vw;
  }

  .Recipe__heading {
    font-size: 8vw;
  }

  .Recipe__heading_small {
    font-size: 5vw;
    margin: 5.6vw 0 6.3vw;
  }
}