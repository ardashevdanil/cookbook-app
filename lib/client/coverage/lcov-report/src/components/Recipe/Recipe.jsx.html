<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Recipe/Recipe.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Recipe</a> Recipe.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react';
import './Recipe.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-regular-svg-icons';
import { faUtensils } from '@fortawesome/free-solid-svg-icons';
&nbsp;
function Recipe({ content }) {
  const ingridients = content.ingridients.split('\\').map((item, index) =&gt; (
    &lt;p key={index}&gt;
      {item}
    &lt;/p&gt;
  ));
&nbsp;
  return (
    &lt;div className="Recipe"&gt;
      &lt;div className="Recipe__img"&gt;
        &lt;img
          alt={content.name}
          src={content.img}
        /&gt;
      &lt;/div&gt;
      &lt;div className="Recipe__description"&gt;
        &lt;div className="Recipe__heading"&gt;
          {content.name}
        &lt;/div&gt;
        &lt;div className="Recipe__stats"&gt;
          &lt;FontAwesomeIcon icon={faClock} /&gt;
          {content.cookingTime}
          mins&amp;#160;
          &lt;FontAwesomeIcon icon={faUtensils} /&gt;
          &amp;#160;
          {content.difficult}
        &lt;/div&gt;
        &lt;div className="Recipe__text"&gt;
          {ingridients}
          &lt;div className="Recipe__heading Recipe__heading_small"&gt;
            Directions
          &lt;/div&gt;
          {content.directions}
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
}
&nbsp;
export default Recipe;
&nbsp;
Recipe.propTypes = {
  content: PropTypes.shape({
    cookingTime: PropTypes.number.isRequired,
    difficult: PropTypes.string.isRequired,
    directions: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    ingridients: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
