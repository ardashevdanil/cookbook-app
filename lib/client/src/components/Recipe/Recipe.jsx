import PropTypes from 'prop-types';
import React from 'react';
import './Recipe.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-regular-svg-icons';
import { faUtensils } from '@fortawesome/free-solid-svg-icons';

function Recipe({ content }) {
  const ingridients = content.ingridients.split('\\').map((item, index) => (
    <p key={index}>
      {item}
    </p>
  ));

  return (
    <div className="Recipe">
      <div className="Recipe__img">
        <img
          alt={content.name}
          src={content.img}
        />
      </div>
      <div className="Recipe__description">
        <div className="Recipe__heading">
          {content.name}
        </div>
        <div className="Recipe__stats">
          <FontAwesomeIcon icon={faClock} />
          {content.cookingTime}
          mins&#160;
          <FontAwesomeIcon icon={faUtensils} />
          &#160;
          {content.difficult}
        </div>
        <div className="Recipe__text">
          {ingridients}
          <div className="Recipe__heading Recipe__heading_small">
            Directions
          </div>
          {content.directions}
        </div>
      </div>
    </div>
  );
}

export default Recipe;

Recipe.propTypes = {
  content: PropTypes.shape({
    cookingTime: PropTypes.number.isRequired,
    difficult: PropTypes.string.isRequired,
    directions: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    img: PropTypes.string.isRequired,
    ingridients: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};
