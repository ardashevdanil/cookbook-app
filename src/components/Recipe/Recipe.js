import React from 'react';
import './Recipe.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faClock } from '@fortawesome/free-regular-svg-icons';
import { faUtensils } from '@fortawesome/free-solid-svg-icons';

import exampleImg from './exampleImg.png'

function Recipe() {

  const ingridients = fakeData.ingridients.map( (item, key) => {
    return <p key={ key }>{ item }</p>
  })

  return(
    <div className='Recipe'>
      <div className='Recipe__img'>
        <img src={ exampleImg } />
      </div>
      <div className='Recipe__description'>
        <div className='Recipe__heading'>
          { fakeData.name }
        </div>
        <div className='Recipe__stats'>
          <FontAwesomeIcon icon={ faClock } />
          { fakeData.cookingTime }mins&#160;
          <FontAwesomeIcon icon={ faUtensils } />&#160;
          { fakeData.difficult }
        </div>
        <div className='Recipe__text'>
          { ingridients }
          <div className='Recipe__heading Recipe__heading_small'>
            Directions
          </div>
          { fakeData.directions }
        </div>
      </div>
    </div>
  );
}

export default Recipe;

const fakeData = {
  name: 'Bananas Foster Ice Cream Cake',
  cookingTime: 15,
  difficult: 'Easy',
  ingridients:[
    `1 (8 ounce) container frozen whipped tapping, thawed`,
    `20 chocolate round wafers, divided`,
    `1 (7 1/4 ounce) bottle Smucker's Hot Dark Chocolat
      Microwaveable Topping, divided`
  ],
  directions: `ARRANGE 8 to 9 ice cream sandwiches in 9-inch
    square pan, cutting to form on even layer. Spread with half
    of whipped tapping. Microwave 1 cup of chocolate
    topping into small microwave-safe bowl on HIGH for 30 seconds
    \nDrizzle evenly over wafers.\nTOP with remaining 8 to 9 ice cream
    sandwiches to form an even layer.\n
    ARRANGE 8 to 9 ice cream sandwiches in 9-inch
    square pan, cutting to form on even layer. Spread with half
    of whipped tapping. Microwave 1 cup of chocolate
    topping into small microwave-safe bowl on HIGH for 30 seconds
    \nDrizzle evenly over wafers.\nTOP with remaining 8 to 9 ice cream
    sandwiches to form an even layer.\n`,
  img: exampleImg,
}