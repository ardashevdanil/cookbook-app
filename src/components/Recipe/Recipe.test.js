import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import Recipe from './Recipe';

Enzyme.configure({ adapter: new Adapter() });

describe('test Recipe component', () => {
  const component = Enzyme.shallow(
    <Recipe />
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });
})