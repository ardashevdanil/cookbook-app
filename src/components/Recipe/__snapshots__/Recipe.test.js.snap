// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Recipe component renders correctly 1`] = `
"<div className=\\"Recipe\\">
  <div className=\\"Recipe__img\\">
    <img src=\\"exampleImg.png\\" />
  </div>
  <div className=\\"Recipe__description\\">
    <div className=\\"Recipe__heading\\">
      Bananas Foster Ice Cream Cake
    </div>
    <div className=\\"Recipe__stats\\">
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
      15
      mins 
      <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} transform={{...}} />
       
      Easy
    </div>
    <div className=\\"Recipe__text\\">
      <p>
        1 (8 ounce) container frozen whipped tapping, thawed
      </p>
      <p>
        20 chocolate round wafers, divided
      </p>
      <p>
        1 (7 1/4 ounce) bottle Smucker&#39;s Hot Dark Chocolat
              Microwaveable Topping, divided
      </p>
      <div className=\\"Recipe__heading Recipe__heading_small\\">
        Directions
      </div>
      ARRANGE 8 to 9 ice cream sandwiches in 9-inch
          square pan, cutting to form on even layer. Spread with half
          of whipped tapping. Microwave 1 cup of chocolate
          topping into small microwave-safe bowl on HIGH for 30 seconds
          
      Drizzle evenly over wafers.
      TOP with remaining 8 to 9 ice cream
          sandwiches to form an even layer.
      
          ARRANGE 8 to 9 ice cream sandwiches in 9-inch
          square pan, cutting to form on even layer. Spread with half
          of whipped tapping. Microwave 1 cup of chocolate
          topping into small microwave-safe bowl on HIGH for 30 seconds
          
      Drizzle evenly over wafers.
      TOP with remaining 8 to 9 ice cream
          sandwiches to form an even layer.
      
    </div>
  </div>
</div>"
`;
